Validator/Constraints/PhoneNumberValidator.php

```diff
@@ -117,9 +122,18 @@ public function validate($value, Constraint $constraint)
      */
     private function addViolation($value, Constraint $constraint)
     {
-        $this->context->addViolation(
-            $constraint->getMessage(),
-            array('{{ type }}' => $constraint->getType(), '{{ value }}' => $value)
-        );
+        /** @var \Misd\PhoneNumberBundle\Validator\Constraints\PhoneNumber $constraint */
+        if ($this->context instanceof ExecutionContextInterface) {
+            $this->context->buildViolation($constraint->getMessage())
+                ->setParameter('{{ type }}', $constraint->getType())
+                ->setParameter('{{ value }}', $this->formatValue($value))
+                ->setCode(PhoneNumber::INVALID_PHONE_NUMBER_ERROR)
+                ->addViolation();
+        } else {
+            $this->context->addViolation($constraint->getMessage(), array(
+                '{{ type }}' => $constraint->getType(),
+                '{{ value }}' => $value
+            ));
```

Comment:
```php
$this->buildViolation($constraint->getMessage())
    ->setParameter('{{ type }}', $constraint->getType())
    ->setParameter('{{ value }}', $this->formatValue($value))
    ->setCode(PhoneNumber::INVALID_PHONE_NUMBER_ERROR)
    ->addViolation();
```

Like this?

Reply:
The exact same code is a few lines above the one you reviewed but as this bundle still supports Symfony 2.3 I have to support both ways of adding a violation to the context

Reply:
Not completely, my example uses the `buildValidation()` method from the base `ConstraintValidator` class and should only be used in the `else` branch of the condition (you can, for example, take a look at the `ExpressionValidator` class in the Symfony core to see how it's done there).

Reply:
The `ConstraintValidator::buildValidation()` method is not available in Symfony 2.3